app/src/features/apiClient/screens/PostmanImporterView/components/PostmanImporter/PostmanImporter.tsx

```diff
@@ -0,0 +1,328 @@
+              collectionsCount.current += collections.length;
+            }
+          }
+        });
+
+        trackImportFromPostmanDataProcessed(collectionsCount.current, processedRecords.environments.length);
+
+        setProcessedFileData(processedRecords);
+        setIsDataProcessed(true);
+      })
+      .catch((error) => {
+        setImportError(error.message);
+        setIsDataProcessed(false);
+      })
+      .finally(() => {
+        setIsDataProcessing(false);
+      });
+  }, []);
+
+  const handleImportEnvironments = useCallback(async () => {
+    try {
+      const importPromises = processedFileData.environments.map(async (env) => {
+        const newEnvironment = await addNewEnvironment(`(Imported) ${env.name}`);
+        if (newEnvironment) {
+          await setVariables(newEnvironment.id, env.variables);
+          return true;
+        }
+        return false;
+      });
+
+      const results = await Promise.all(importPromises);
+      return results.filter(Boolean).length;
+    } catch (error) {
+      Logger.error("Postman data import failed:", error);
+      throw error;
+    }
+  }, [addNewEnvironment, setVariables, processedFileData.environments]);
+
+  const handleImportCollectionsAndApis = useCallback(async () => {
+    let importedCollectionsCount = 0;
+
+    const collections = processedFileData.apiRecords.filter((record) => record.type === RQAPI.RecordType.COLLECTION);
+    const apis = processedFileData.apiRecords.filter((record) => record.type === RQAPI.RecordType.API);
+
+    const collectionsPromises = collections.map(async (collection) => {
+      const collectionToImport = {
+        ...collection,
+        name: `(Imported) ${collection.name}`,
+      };
+      delete collectionToImport.id;
+
+      const newCollection = await upsertApiRecord(user?.details?.profile?.uid, collectionToImport, workspace?.id);
+      onSaveRecord(newCollection.data);
+      importedCollectionsCount++;
+      return {
+        oldId: collection.id,
+        newId: newCollection.data.id,
+      };
+    });
+
+    const collectionsResult = await Promise.all(collectionsPromises);
+    const oldToNewCollectionIds: Record<string, { newId: string }> = collectionsResult.reduce(
+      (result, details) => ({
+        ...result,
+        [details.oldId]: { newId: details.newId },
+      }),
+      {}
+    );
+
+    // Import APIs with updated collection IDs
+    const apisPromises = apis.map(async (api) => {
+      const apiToImport = { ...api };
+      delete apiToImport.id;
+
+      const newCollectionId = oldToNewCollectionIds[api.collectionId]?.newId;
+      if (!newCollectionId) {
+        throw new Error(`Failed to find new collection ID for API: ${api.name || api.id}`);
+      }
+
+      const updatedApi = { ...apiToImport, collectionId: newCollectionId };
+      const newApi = await upsertApiRecord(user.details?.profile?.uid, updatedApi, workspace?.id);
```

Comment:
Here to, what if some of the updates fail?

Reply:
Added warning toast for partial imports